packages/pros/src/lib.rs

```diff
@@ -44,372 +45,73 @@
-/// }
-/// #[async_trait]
-/// impl AsyncRobot for ExampleRobot {
-///     async fn opcontrol(&mut self) -> pros::Result {
-///         println!("Hello, world! {}", self.x);
-///         Ok(())
-///     }
-/// }
-/// impl ExampleRobot {
-///     pub fn new() -> Self {
-///        Self { x: 5 }
-///    }
-/// }
-/// async_robot!(ExampleRobot, ExampleRobot::new());
-#[macro_export]
-macro_rules! async_robot {
-    ($rbt:ty) => {
-        $crate::__gen_async_exports!($rbt);
-
-        #[no_mangle]
-        extern "C" fn initialize() {
-            unsafe {
-                ROBOT = Some(Default::default());
-            }
-        }
-    };
-    ($rbt:ty, $init:expr) => {
-        $crate::__gen_async_exports!($rbt);
-
-        #[no_mangle]
-        extern "C" fn initialize() {
-            unsafe {
-                ROBOT = Some($init);
-            }
-        }
-    };
-}
-
-/// Allows your sync robot code to be executed by the pros kernel.
-/// If your robot struct implements Default then you can just supply this macro with its type.
-/// If not, you can supply an expression that returns your robot type to initialize your robot struct.
-/// The code that runs to create your robot struct will run in the initialize function in PROS.
-///
-/// Example of using the macro with a struct that implements Default:
-/// ```rust
-/// use pros::prelude::*;
-/// #[derive(Default)]
-/// struct ExampleRobot;
-/// impl SyncRobot for ExampleRobot {
-///    asnyc fn opcontrol(&mut self) -> pros::Result {
-///       println!("Hello, world!");
-///      Ok(())
-///   }
-/// }
-/// sync_robot!(ExampleRobot);
-/// ```
-///
-/// Example of using the macro with a struct that does not implement Default:
-/// ```rust
-/// use pros::prelude::*;
-/// struct ExampleRobot {
-///    x: i32,
-/// }
-/// impl SyncRobot for ExampleRobot {
-///     async fn opcontrol(&mut self) -> pros::Result {
-///         println!("Hello, world! {}", self.x);
-///         Ok(())
-///     }
-/// }
-/// impl ExampleRobot {
-///     pub fn new() -> Self {
-///        Self { x: 5 }
-///    }
-/// }
-/// sync_robot!(ExampleRobot, ExampleRobot::new());
-#[macro_export]
-macro_rules! sync_robot {
-    ($rbt:ty) => {
-        $crate::__gen_sync_exports!($rbt);
-
-        #[no_mangle]
-        extern "C" fn initialize() {
-            unsafe {
-                ROBOT = Some(Default::default());
-            }
-        }
-    };
-    ($rbt:ty, $init:expr) => {
-        $crate::__gen_sync_exports!($rbt);
-
-        #[no_mangle]
-        extern "C" fn initialize() {
-            unsafe {
-                ROBOT = Some($init);
-            }
-        }
-    };
-}
+#[cfg(feature = "async")]
+pub use pros_async as async_runtime;
+#[cfg(feature = "core")]
+pub use pros_core as core;
+#[cfg(feature = "devices")]
+pub use pros_devices as devices;
+#[cfg(feature = "math")]
+pub use pros_math as math;
+#[cfg(feature = "panic")]
+pub use pros_panic as panic;
+#[cfg(feature = "sync")]
+pub use pros_sync as sync;
 
```

Comment:
Maybe consider re-exporting pros-sys here to fit with the pattern of `pros::modulename`

Reply:
I considered it, but I think that it's best not to expose `pros-sys`.